### PR TITLE
Fixed PackagePath for files without extension

### PR DESCRIPTION
**Pull request by jp2masa**
## Bug
Fixes NuGet/Home#5898

## Testing/Validation
Tests Added: Yes

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs [hunk 1]
```diff
@@ -506,7 +507,8 @@ private void InitNuspecOutputPath(IPackTaskRequest<IMSBuildItem> request, PackAr
                 var currentPath = targetPath;
                 var fileName = Path.GetFileName(sourcePath);
                 if (!Path.GetExtension(fileName)
-                    .Equals(Path.GetExtension(targetPath), StringComparison.OrdinalIgnoreCase))
+                    .Equals(Path.GetExtension(targetPath), StringComparison.OrdinalIgnoreCase)
+                    || Path.GetExtension(fileName) == String.Empty)
```
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs]: change to string.IsNullOrEmpty(Path.GetExtension(fileName)) and make that to be the first condition check

**Comment by emgarten** [comment]
@rohit21agrawal is this change good to go in?

**Comment by rohit21agrawal** [comment]
Yes, running tests now

**Comment by emgarten** [comment]
@rohit21agrawal not sure what the test status is on this, please take a look when you are back.

**Pull request merged**